### PR TITLE
Add multi-environment support for AKS

### PR DESCRIPTION
**Pull request by mjura**
Fixes: https://github.com/rancher/aks-operator/issues/98

(cherry picked from commit 41a68e8345dc49338cf7cc42a93ad09d9afd7318)

### CODE REVIEW

**Review by richardcase — APPROVED** [review]
Seems good to me.

**Review by a-blender — APPROVED** [review]
How was this tested?

File: pkg/aks/client.go [hunk 1]
```diff
@@ -152,3 +164,16 @@ func NewClusterClient(cred *Credentials) (*containerservice.ManagedClustersClien
 
 	return &client, nil
 }
+
+func GetEnvironment(env string) azure.Environment {
```
[the remark this answers is not in the document]
Comment (mjura) [reply pkg/aks/client.go:168]: I have to check it

**Comment by mjura** [comment]
> How was this tested?

Thank you for asking. This change wasn't tested or it was tested on **AzurePublicCloud**. We need someone who has access to gov cloud to test it

File: pkg/aks/client.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexander-demicev) [reply pkg/aks/client.go:168]: azure stack will require a separate implementation, it has a different API that doesn't support standard lb, spot instances, etc. We'll have to treat azure stack as a new provider.

**Review by alexander-demicev — APPROVED** [review]
lgtm
testing on govcloud will require someone with access to it or a partnership with Microsoft, I know they can provide a `simulator`.

**Pull request merged**